src/pybind/rados/rados.pyx

```diff
@@ -2122,6 +2125,24 @@ cdef class WriteOp(object):
         with nogil:
             rados_write_op_truncate(self.write_op,  _offset)
 
+    @requires(('to_write', bytes), ('write_len', int), ('offset', int))
+    def writesame(self, to_write, write_len, offset=0):
```

Comment:
it would be nice if you can keep the order of the functions in alphabetical order like done in function declaration above, which would mean `rados_writesame` being defined after `rados_write_full`